x/posts/internal/keeper/handler.go

```diff
@@ -144,18 +143,33 @@ func handleMsgAddPostReaction(ctx sdk.Context, keeper Keeper, msg types.MsgAddPo
 		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("post with id %s not found", msg.PostID))
 	}
 
-	// Get the reaction value
-	reactionValue := strings.ReplaceAll(msg.Reaction, "️", "")
+	var reactionShortcode string
+	var reactionValue string
+	// Parse reaction adding the variation selector-16 to let the emoji being readable
+	parsedReaction := strings.ReplaceAll(msg.Reaction, "️", "")
 
 	// Check if the reaction is an emoji
-	if emojiReact, err := emoji.LookupEmoji(reactionValue); err == nil {
-		reactionValue = emojiReact.Shortcodes[0]
-
-		// nolint: errcheck - We don't care if the reaction is already registered
-		_ = registerReaction(ctx, keeper, reactionValue, post.Subspace, msg.Reaction, types.ModuleAddress)
+	if emojiReact, err := emoji.LookupEmoji(parsedReaction); err == nil {
+		reactionShortcode = emojiReact.Shortcodes[0]
+		reactionValue = emojiReact.Value
+	} else { //if not, check if the shortcode is related to any emoji
+		emo, err := emoji.LookupEmojiByCode(parsedReaction)
+		if err != nil { // if no emoji is related to it, check if the shortcode has been registered before
+			regReaction, exist := keeper.GetRegisteredReaction(ctx, parsedReaction, post.Subspace)
+			if !exist { // if it has not been registered yet, no one can use it to react to the post
+				return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+					fmt.Sprintf("reaction with short code %s isn't registered yet and can't be used to react to the post with ID %s and subspace %s, please register it before use",
+						parsedReaction, post.PostID, post.Subspace))
+			}
+			reactionValue = regReaction.Value
+			reactionShortcode = regReaction.ShortCode
+		} else {
+			reactionValue = emo.Value
+			reactionShortcode = emo.Shortcodes[0]
+		}
```

Comment:
I'm not a big fan of this piece of code's structure. I think it can be improve by doing something like 

```golang
if emojiReact, err := emoji.LookupEmoji(parsedReaction); err == nil {
  // The reaction was an emoji Unicode character
} else if emojiReact, err := emoji.LookupEmojiByCode(parsedReaction); err == nil {
  // The reaction was an emoji shotcode
} else {
  // The reaction was a shortcode that should be registered
}
```

I think it would be more easy to read. Also, since this entire piece of code is used as well inside the `handleMsgRemovePostReaction`, I would avoid duplicating it. To do this, we should move it into an external function.

Reply:
yeah I was not so conviced as well but I couldn’t think a better way to do that in that moment so I was hoping for some advice from you!